pkg/util/encryptionconfig/config.go

```diff
@@ -0,0 +1,78 @@
+package encryptionconfig
+
+import (
+	"context"
+	"fmt"
+	"os"
+
+	"github.com/rancher/backup-restore-operator/pkg/util"
+	v1 "github.com/rancher/wrangler/v3/pkg/generated/controllers/core/v1"
+
+	"github.com/sirupsen/logrus"
+	v1core "k8s.io/api/core/v1"
+	v2 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	k8sEncryptionconfig "k8s.io/apiserver/pkg/server/options/encryptionconfig"
+	storagevalue "k8s.io/apiserver/pkg/storage/value"
+	"k8s.io/apiserver/pkg/storage/value/encrypt/identity"
+)
+
+type contextKey string
+
+var tempConfigPathKey = contextKey("tmpConfigPath")
+
+const EncryptionProviderConfigKey = "encryption-provider-config.yaml"
+
+func GetEncryptionConfigSecret(secrets v1.SecretController, encryptionConfigSecretName string) (*v1core.Secret, error) {
+	// EncryptionConfig secret ns is hardcoded to ns of controller in chart's ns
+	// kubectl create secret generic test-encryptionconfig --from-file=./encryption-provider-config.yaml
+	logrus.Infof("Get encryption config from namespace %v", util.GetChartNamespace())
+	encryptionConfigSecret, err := secrets.Get(util.GetChartNamespace(), encryptionConfigSecretName, v2.GetOptions{})
+	if err != nil {
+		return nil, err
+	}
+
+	return encryptionConfigSecret, nil
+}
+
+func GetEncryptionTransformersFromSecret(ctx context.Context, encryptionConfigSecret *v1core.Secret) (k8sEncryptionconfig.StaticTransformers, error) {
+	fileHandle, err := PrepareEncryptionConfigSecretTempConfig(encryptionConfigSecret)
+	// we defer file removal till here to ensure it's around for all of PrepareEncryptionTransformersFromConfig
+	defer os.Remove(EncryptionProviderConfigKey)
+	if err != nil {
+		return nil, err
+	}
+	ctx = context.WithValue(ctx, tempConfigPathKey, fileHandle.Name())
+	return PrepareEncryptionTransformersFromConfig(ctx, EncryptionProviderConfigKey)
+}
+
+func PrepareEncryptionConfigSecretTempConfig(encryptionConfigSecret *v1core.Secret) (*os.File, error) {
+	encryptionConfigBytes, ok := encryptionConfigSecret.Data[EncryptionProviderConfigKey]
+	if !ok {
+		return nil, fmt.Errorf("no encryptionConfig provided")
+	}
+	err := os.WriteFile(EncryptionProviderConfigKey, encryptionConfigBytes, os.ModePerm)
+	if err != nil {
+		return nil, err
+	}
+
+	// Open the file for reading (or other operations) and return the handle
+	file, err := os.Open(EncryptionProviderConfigKey)
+	if err != nil {
+		return nil, err
+	}
+
```

Comment:
I didn't notice this before, but you we should be careful about opening files inside a function and expecting others to close said file. 

Can lead to concurrent closes or file descriptor leaks. 
We should look into alternative return types instead of `os.File`. I also don't see this being used by anything other than the tests at the moment. 

Will we need this inside the controller logic?